src/device.cpp

```diff
@@ -426,6 +432,23 @@ void cvk_device::init_compiler_options() {
     if (device_16bit_storage_features().storagePushConstant16 == VK_FALSE) {
         m_device_compiler_options += " -no-16bit-storage=pushconstant ";
     }
+    std::vector<std::string> roundingModeRTE;
+    if (m_float_controls_properties.shaderRoundingModeRTEFloat16) {
+        roundingModeRTE.push_back("16");
+    }
+    if (m_float_controls_properties.shaderRoundingModeRTEFloat32) {
+        roundingModeRTE.push_back("32");
+    }
+    if (m_float_controls_properties.shaderRoundingModeRTEFloat64) {
+        roundingModeRTE.push_back("64");
+    }
+    if (roundingModeRTE.size() > 0) {
+        m_device_compiler_options += " -rounding-mode-rte=";
+        std::stringstream stream;
+        std::copy(roundingModeRTE.begin(), roundingModeRTE.end(),
+                  std::ostream_iterator<std::string>(stream, ","));
```

Comment:
This results in a trailing comma which clspv does not accept. You're apparently going to have to wait for  https://en.cppreference.com/w/cpp/experimental/ostream_joiner to make it into a future standard for a solution from the standard library :P. I'm guessing it passed the CI because swiftshader does not support float controls.

Reply:
fixed